pallets/rewards/src/lib.rs

```diff
@@ -0,0 +1,291 @@
+    /// Weight information for extrinsic of the sto pallet.
+    type WeightInfo: WeightInfo;
+}
+
+pub trait WeightInfo {
+    fn claim_itn_reward() -> Weight;
+    fn set_itn_reward_status() -> Weight;
+}
+
+/// Represents an Itn reward's status.
+#[derive(Decode, Encode, Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum ItnRewardStatus {
+    Unclaimed(Balance),
+    Claimed,
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> {
+        /// Address was not found in the list of Itn addresses.
+        UnknownItnAddress,
+        /// Itn reward was already claimed.
+        ItnRewardAlreadyClaimed,
+        /// Provided signature was invalid.
+        InvalidSignature,
+        /// Balance can not be converted to a primitive.
+        UnableToCovertBalance,
+    }
+}
+
+decl_storage! {
+    trait Store for Module<T: Config> as Rewards {
+        /// Map of (Itn Address `AccountId`) -> (Reward `ItnRewardStatus`).
+        pub ItnRewards get(fn itn_rewards): map hasher(blake2_128_concat) T::AccountId
+            => Option<ItnRewardStatus>;
+    }
+    add_extra_genesis {
+        config(itn_rewards): Vec<(T::AccountId, Balance)>;
+        build(|config: &GenesisConfig<T>| {
+            for (account, balance) in &config.itn_rewards {
+                <Module::<T>>::base_set_itn_reward_status(account, ItnRewardStatus::Unclaimed(*balance));
+            }
+        });
+    }
+}
+
+decl_event! {
+    pub enum Event<T>
+    where
+        AccountId = <T as frame_system::Config>::AccountId,
+    {
+        /// Itn reward was claimed.
+        ItnRewardClaimed(AccountId, Balance),
+    }
+}
+
+decl_module! {
+    pub struct Module<T: Config> for enum Call where origin: <T as frame_system::Config>::Origin {
+        type Error = Error<T>;
+
+        fn deposit_event() = default;
+
+        /// Claim an ITN reward.
+        ///
+        /// ## Arguments
+        /// * `itn_address` specifying the awarded address on ITN.
+        /// * `signature` authenticating the claim to the reward.
+        ///    The signature should contain `reward_address` followed by the suffix `"claim_itn_reward"`,
+        ///    and must have been signed by `itn_address`.
+        ///
+        /// # Errors
+        /// * `InsufficientBalance` - Itn rewards has insufficient funds to issue the reward.
+        /// * `InvalidSignature` - `signature` had an invalid signer or invalid message.
+        /// * `ItnRewardAlreadyClaimed` - Reward issued to the `itn_address` has already been claimed.
+        /// * `UnknownItnAddress` - `itn_address` is not in the rewards table and has no reward to be claimed.
+        #[weight = <T as Config>::WeightInfo::claim_itn_reward()]
+        pub fn claim_itn_reward(origin, reward_address: T::AccountId, itn_address: T::AccountId, signature: T::OffChainSignature) -> DispatchResult {
+            ensure_none(origin)?;
+            //TODO(Connor): Check cdd claim.
```

Comment:
What's the plan here?

Reply:
TBD. Adam had suggested to switch the underlying call of Currency::transfer to the version which does the CDD check.

Reply:
Going to solve this in a new PR.